services/BaseService.js

```diff
@@ -57,6 +57,15 @@ export default class BaseService {
       }
       throw Error('fail to insert data');
     } catch (error) {
+      if (error.name === 'SequelizeUniqueConstraintError') {
```

Comment:
untuk validasi unique email dan username bukan generic, dalam kasus ini hanya ada pada user register. saya rasa boleh mengubah baseService dengan memperhatikan prinsip `close for modification open for extension` mungkin membuat method baru di eg: `register()` di `UserService`. karena yang ada di `BaseService` tujuannya bersifat generic dan dipakai semua service ke depannya.

Reply:
ok i see. got it.